WPMedia/fusion-news-theme-blocks
# THEMES-986: Convert header nav chain tests

## Description

This PR converts the header nav chain tests from using Enzyme to using React Testing Library. Some tests were removed because they were redundant, or were testing non-existent functionality.

## Jira Ticket

- [THEMES-986](https://arcpublishing.atlassian.net/browse/THEMES-986)

## Acceptance Criteria

- Convert tests using enzyme to RTL

## Test Steps

1. Checkout this branch `git checkout THEMES-986`
2. Run the block repo's tests: `npm run test`
3. All tests for the header nav chain should pass.

## Author Checklist

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[THEMES-986]: https://arcpublishing.atlassian.net/browse/THEMES-986?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
